app/javascript/shared/components/keyboardShortcuts.jsx

```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import { h } from 'preact';
+import { useEffect } from 'preact/hooks';
+
+/**
+ * Component that can be added to the dom to listen 
+ * for keyboard presses
+ * 
+ * @example
+ * const shortcuts = {
+ *   "ctrl+alt+KeyG": (e) => {
+ *     e.preventDefault(); 
+ *     alert("Control Alt G has been pressed")
+ *   }
+ * }
+ * <KeyboardShortcuts shortcuts={shortcuts} />
+ * 
+ */
+const KeyboardShortcuts = ({ shortcuts, ...props }) => {
```

Comment:
This is a component that returns `null` always. That's OK to do and valid, but given that this component is just a `useEffect`, consider making a custom hook instead.

I made a quick proof of concept here, https://codesandbox.io/s/alternate-suggestion-for-httpsgithubcomforemforempull10713-w5fro?file=/src/useKeyboardShortcuts.js

If we need to support class based components as well, you can do what I did for drag and drop, e.g. https://github.com/forem/forem/blob/master/app/javascript/utilities/dragAndDrop.js. I provide a hook and a component.

Thoughts?

Reply:
Nice! I like the hook/component combo. With that said, I think the `KeyboardShortcuts` component's difference will be that it should not necessarily "wrap" anything (or obligatory have children) as we want the event listeners on `document` most of the time and not only on the children elements.

Note: I'll WIP #10468 for now while we settle on this, and update it as soon as it's merged

Reply:
Thanks for the review! I've moved it over to a hook and added a component that uses that hook.